quic/s2n-quic-core/src/event/metrics.rs

```diff
@@ -0,0 +1,17 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
```

Comment:
```suggestion
/// A Recorder should arrange to emit the properties and counters on Drop into some output.
```

Maybe?

I wonder if we should also note in `register_counter` that the counter shouldn't be used after dropping the recorder (i.e., that might be a no-op rather than incrementing a counter still). I don't think it makes sense to have the Counter borrow from the Recorder -- too constraining to manage those lifetimes -- but documenting the intended lifecycle seems like a good idea.